strings/base_error.h

```diff
@@ -426,6 +426,11 @@ WINRT_EXPORT namespace winrt
 
     [[noreturn]] inline __declspec(noinline) void throw_hresult(hresult const result)
     {
+        if (winrt_throw_hresult_handler)
+        {
+            winrt_throw_hresult_handler(0, nullptr, nullptr, _ReturnAddress(), result);
```

Comment:
`_ReturnAddress` is a MSVC extension. For portability, may need to say

```cpp
    void* returnAddress = nullptr;
#if _MSC_VER
    returnAddress = _ReturnAddress();
#elif defined(__GNUC__) // also detects clang, icc, djgpp, ellcc, zapcc
    returnAddress = __builtin_extract_return_addr(__builtin_return_address(0));
#endif
```

Reply:
`to_hresult` was already using `_ReturnAddress` but yes not very portable...

Reply:
I was following the pattern established by the similar `to_hresult` extension point with the expectation that it was already correct/acceptable.

(For what it's worth the MSVC version of your snippet is a lot more readable than the else block :))